internal/sysfs/adapter_test.go

```diff
@@ -168,18 +169,69 @@ func TestAdapt_HackedWrites(t *testing.T) {
 // when a fs.FS returns an os.File or methods we use from it.
 type MaskOsFS struct {
 	Fs fs.FS
+
+	// ZeroIno helps us test stat with sys.Stat_t work.
+	ZeroIno bool
 }
 
-func (f *MaskOsFS) Open(name string) (fs.File, error) {
-	if f, err := f.Fs.Open(name); err != nil {
+// Open implements the same method as documented on fs.FS
+func (fs *MaskOsFS) Open(name string) (fs.File, error) {
+	if f, err := fs.Fs.Open(name); err != nil {
 		return nil, err
 	} else if osF, ok := f.(*os.File); !ok {
 		return nil, fmt.Errorf("input not an os.File %v", osF)
+	} else if fs.ZeroIno {
+		return &zeroInoOsFile{osF}, nil
 	} else {
 		return struct{ methodsUsedByFsAdapter }{osF}, nil
 	}
 }
 
+// zeroInoOsFile wraps an os.File to override functions that can return
+// fs.FileInfo.
+type zeroInoOsFile struct{ *os.File }
```

Comment:
I chose to not make a top-level example for this, as we can point the few who do this to the unit test. Reason being it is complicated and godoc examples can't render anything outside the function body. Instead, we should document the real fs once we expose it, shortly.